docs/source/blueprints/BP003.rst

```diff
@@ -0,0 +1,549 @@
+BP003
+#####
+
+:Number: BP003
+:Title: N(ext)Runner Task Phases
+:Author: Cleber Rosa <crosa@redhat.com>
+:Discussions-To: avocado-devel@redhat.com
+:Reviewers:
+:Created: 20-July-2020
+:Type: Architecture Blueprint
+:Status: Draft
+
+.. contents:: Table of Contents
+
+TL;DR
+*****
+
+The N(ext) Runner has been used as Avocado's runner for selftests for
+over a year.  The implementation used is based on the ``avocado nrun``
+command, that is, outside of the Avocado's traditional ``avocado run``
+entrypoint.  Under the hood, it means that the N(ext) Runner is not
+integrated well enough with an Avocado Job.
+
+A partial implementation of the N(ext) Runner integration with an
+Avocado Job is available at ``avocado/plugins/runner_nrunner.py`` but
+it has a number limitations.
+
+The N(ext) Runner executes tests as :class:`Tasks
+<avocado.core.nrunner.Task>`.  This blueprint describes the phases
+that a Task can be in throughout its life-cycle, and how the handling
+of these phases or states, will power the tests execution mechanism
+within the context of an Avocado Job.
+
+Motivations
+***********
+
+1. Explain how, with the distinguishing N(ext) Runner features, new
+   requirements are introduced with regards to its integration with an
+   Avocado Job.
+
+2. Propose an architecture based on the phases that an N(ext) Runner
+   Task can go through while running under an Avocado Job.
+
+3. Describe how the proposed architecture can power an implementation
+   suitable for the next Avocado LTS release (82.0), having feature
+   completeness when compared to the current runner, while still
+   making its distinguishing features available to users who opt in.
+
```

Comment:
What I miss here is... advantages over what?  There's no other architecture being discussed or proposed so that it can be compared with this proposal.

Reply:
advantages of having one! Why did you decide to propose this one?

Reply:
After a conversation with @beraldoleal it was understood that these are the goals of this BluePrint, and not of what's proposed by the BluePrint.  I'm moving points 2-6 to another section called "Goals of this BluePrint".